### PR TITLE
[CARBONDATA-1651] [Supported Boolean Type When Saving DataFrame] Provided Support For Boolean Data Type In CarbonDataFrameWriter

### PR DESCRIPTION
**Pull request by anubhav100**
1.Provided Support For Boolean Data Type In CarbonDataFrameWriter
2.Test Cases are Added For Same

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1047/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1665/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataFrame.scala [hunk 1]
```diff
@@ -70,7 +82,17 @@ class TestLoadDataFrame extends QueryTest with BeforeAndAfterAll {
...
+test("test the boolean data type"){
+  booldf.write
+    .format("carbondata")
+    .option("tableName", "carbon10")
+    .option("tempCSV", "true")
+    .option("compress", "true")
+    .mode(SaveMode.Overwrite)
+    .save()
+  checkAnswer(
+  sql("SELECT * FROM CARBON10"),Seq(Row("anubhav",true),Row("prince",false)))
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataFrame.scala]: change to 
```
checkAnswer(
     sql("SELECT * FROM CARBON10"),
     Seq(Row("anubhav", true), Row("prince", false)))
```

Comment (anubhav100) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestLoadDataFrame.scala]: @jackylk i have done the changes you can review

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1124/

**Comment by anubhav100** [comment]
Retest this

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1720/

**Comment by anubhav100** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1138/

**Comment by anubhav100** [comment]
@jackylk why build is failing i have clean build on machine and also resolved all review comments

**Comment by anubhav100** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1212/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1226/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**